test/powershell/Modules/Microsoft.PowerShell.Management/Start-Process.Tests.ps1

```diff
@@ -169,3 +170,26 @@ Describe "Start-Process tests requiring admin" -Tags "Feature","RequireAdminOnWi
         Get-Content $testdrive\foo.txt | Should -BeExactly $fooFile
     }
 }
+
+Describe "Start-Process" -Tags "Feature" {
+
+    It "UseNewEnvironment parameter should reset environment variables for child process" {
+
+        $PWSH = (Get-Process -Id $PID).MainModule.FileName
+        $outputFile = Join-Path -Path $TestDrive -ChildPath output.txt
+
+        $env:TestEnvVariable | Should -BeNullOrEmpty
+
+        $env:TestEnvVariable = 1
+
+        try {
+            Start-Process $PWSH -ArgumentList '-NoProfile','-Command Write-Output $env:TestEnvVariable' -RedirectStandardOutput $outputFile -Wait
```

Comment:
Use of CreateEnvironmentBlock() with the user token is great.  Did you verify that USERNAME is correct and PATH is complete?

Reply:
We could check USERNAME. Checking PATH it would be that we don't trust native API.

Reply:
Done.